engine/crates/composition/src/validate.rs

```diff
@@ -68,23 +76,37 @@ fn validate_selection(
         ));
     };
 
+    for required_argument in field.arguments().filter(|arg| arg.r#type().is_required()) {
```

Comment:
Oh, one last thing - should you also be checking whether the argument has a default here?  If it's required but has a default then that probably shouldn't be an error

Reply:
(This comment brought to you by [fixing a similar bug in relay](https://github.com/facebook/relay/pull/4647) very recently)

Reply:
Good point! Will do that before merging.